### PR TITLE
New artifact names

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
Spine Model Compiler already handles Java, JavaScript (JS), and Dart. The level of support by language is different, and we want to make improving it easier, as well as introducing new languages, Kotlin in particular.

This PR is the first step in improving the multi-language support under `base`. It introduces new schema for tool artifact names. This PR supersedes #632, reducing introduced changes to (primarily) artifact names. 

Changes to the package structure of Java code will be introduced in the following PRs.

### New naming schema for Module Compiler artifacts

Model Compiler artifact names are composed using the following pattern:
```
 spine-mc-<language>[-<submodule>]
```
where `mc` stands for “Model Compiler”, and `<submodule>` is an optional part of the Model Compiler for the corresponding language. 

For example, `spine-mc-java-protoc` is a part of Model Compiler for Java responsible for integration with Google Protobuf Compiler (`protoc`). And `spine-mc-js` is Mode Compiler for JavaScript.

Correspondingly, _modules_ of the Model Compiler of the `base` project follow the convention:
```
mc-<language>[-<submodule>]
```
... because we add `spine` to the artifact name when publishing to Maven.

### Changes to Gradle plugin names
The following table list changes to the plugin names as they are used in `apply()` calls of a Gradle scripts:

 Previous Name | New Name 
---|---
`io.spine.tools.protobuf-javadoc-plugin` | `io.spine.javadoc-style`
`io.spine.tools.model-compiler` |`io.spine.mc-java`
`io.spine.tools.proto-dart-plugin` | `io.spine.mc-dart`
`io.spine.tools.proto-js-plugin` | `io.spine.mc-js`

We do not repeat the word `tools` in the new plugin IDs, because when they are used, it is obvious that a Gradle plugin is a tool. And we want shorter names there.

### Java package conventions

One change to the package naming was introduced in this PR.
`io.spine.code.gen.java.UuidMethodFactory` moved under `io.spine.tools.java.code`.
Corresponding change was made to `model-compiler.gradle` file in the project (and under `config`).

This change indicates the new convention for naming language-related packages.

If a code-related package is language-specific, but is _NOT_ a part of Model Compiler, it goes as a sub-package under `io.spine.tools`. For example, packages related to Java code would be:

```
io.spine.tools.java.code
io.spine.tools.java.fs
io.spine.tools.java.javadoc
```

If a package _IS_ a part of Model Compiler, it should go under `io.spine.tools.mc`:

```
io.spine.tools.mc.java
io.spine.tools.mc.java-protoc
io.spine.tools.mc.js
io.spine.tools.mc.dart
```

### Other notable changes
* `DependencyConfigurer` was renamed to `ConfigDependency`, and made to act in one `applyTo(Configuration)` call.
* `PreprocessorConfigurer` was renamed to `PreprocessorConfig`, and made to act in one `applyTo(Project)` call.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol @dmdashenkov @yuri-sergiichuk

Gentlemen, I've invited you in addition to all the Comitters, because these name changes are important, and are hardly reversible. So, any additional eye and spotted detail is important. Most of the changes are renaming, so don't waste your time on them. But your attention (sometime starting from Monday) to the PR description and feedback would be appreciated.

File: README.md [hunk 1]
```diff
@@ -4,40 +4,55 @@
...
-* [Spine Error Prone Checks](tools/errorprone-checks) — the custom Error Prone checks for the
-  Spine projects.
+  
+  
+* **Model Compiler for Java** which consists of the following sub-modules:
+  * **[`tools/mc-java`](tools/mc-java)** — a Gradle plugin responsible for enabling Java
+    code generation.
+  * **[`tools/mc-java-protoc`](tools/mc-java-protoc)** — plugs-into Google Protobuf compiler for
+    generating framework-specific Java code.
+  * **[`tools/mc-java-protoc`](tools/mc-java-protoc)** — static code analyzers for Spine-based
```
Comment (dmdashenkov) [inline README.md:23]: Please fix the reference to the module. It's duplicated in the previous list item.

File: README.md [hunk 2]
```diff
@@ -4,40 +4,55 @@
...
+
+* **[`tools/javadoc-style`](tools/javadoc-style)** — a Gradle plugin which processes Javadocs of
+  generated Java files to make them look less ugly.
+* **[`tools/javadoc-filter`](tools/javadoc-filter)** — excludes elements annotated with
   `io.spine.Internal` from the generated doc.
-* [Spine Error Prone Checks](tools/errorprone-checks) — the custom Error Prone checks for the
-  Spine projects.
+  
+  
+* **Model Compiler for Java** which consists of the following sub-modules:
```
Comment (dmdashenkov) [inline README.md:18]: In the Gradle config, there are 4 items, `mc-java-validation` seems to be missing here. Is it an internal module? If not, please include it into this doc.

File: tools/mc-java/src/main/java/io/spine/tools/gradle/compiler/ErrorProneChecksExtension.java [hunk 3]
```diff
@@ -41,10 +43,15 @@
 
     public Severity useValidatingBuilder;
 
+    public static void createIn(Project project) {
```
Comment (dmdashenkov) [inline tools/mc-java/src/main/java/io/spine/tools/gradle/compiler/ErrorProneChecksExtension.java:46]: Please document this piece of public API.

File: tools/mc-java/src/test/java/io/spine/tools/compiler/check/ConfigDependencyTest.java [hunk 4]
```diff
@@ -87,23 +83,22 @@ void notAddSpineCheckDependencyIfItIsNotResolvable() {
...
-
-    private static DependencyConfigurer createFor(Project project) {
         Configuration annotationProcessorConfig = annotationProcessorConfig(project);
-        DependencyConfigurer configurer =
-                DependencyConfigurer.createFor(annotationProcessorConfig);
-        return configurer;
+        ConfigDependency.applyTo(annotationProcessorConfig);
     }
 
+//    private static ConfigDependency applyTo(Project project) {
```
Comment (dmdashenkov) [inline tools/mc-java/src/test/java/io/spine/tools/compiler/check/ConfigDependencyTest.java:90]: Please remove commented code.

**Review by dmdashenkov — COMMENTED** [review]
@alexander-yevsyukov, please see my comments.

File: README.md (hunk 2 above)
Comment (alexander-yevsyukov) [reply README.md:18]: It is going to be internal, but now it's still separate. Added.

**Review by yuri-sergiichuk — COMMENTED** [review]
@alexander-yevsyukov please consider a couple of comments below. Overall LGTM. The new naming conventions look promising.

**Review by yuri-sergiichuk — APPROVED** [review]

**Pull request merged**